src/subprocess-posix.cc

```diff
@@ -123,6 +127,12 @@ bool Subprocess::Start(SubprocessSet* set, const string& command) {
   if (err != 0)
     Fatal("posix_spawn: %s", strerror(err));
 
+  if(priority != INT_MIN) {
+    err = setpriority(PRIO_PROCESS, pid_, priority);
```

Comment:
there is a race condition where the spawned process may have already completed before this code is run. The call would return an error and this would crash Ninja unexpectedly.

Can you try to set the priority before the posix_spawn() call with posix_spawnattr_setschedparam() instead?

Reply:
Somehow, using posix_spawnattr_setschedparam doesn't seem to work. When checking in htop I don't see any change in niceness of the spawned processes.